netbox_netprod_importer/vendors/cisco/ios.py

```diff
@@ -78,3 +107,138 @@ def get_detailed_cdp_neighbours(self):
+                if inf_mode.get("interface"):
+                    interfaces_mode[inf_mode["interface"]] = inf_mode
+
+            self.cache["mode"] = interfaces_mode
+
+        return self.cache["mode"]
+
+    def get_vlans(self):
+        """
+        Napalm does not support vlan
+        but there are issues https://github.com/napalm-automation/napalm/issues/927
+        PR: https://github.com/napalm-automation/napalm/pull/948
+
+        :return vlans:
+            vlan_id, {
+                "name": vlan name,
+                "interfaces": list interfaces dict
+            }
+        """
+        interfaces = self.device.get_interfaces()
+        interface_dict = {}
+        for interface, data in interfaces.items():
+            interface_dict[self.get_abrev_if(interface)] = interface
+
+        command = "show vlan all-ports"
+        output = self.device.cli([command])[command]
+        if output.find("Invalid input detected") >= 0:
+            yield from self._get_vlan_from_id(interface_dict)
+        else:
+            yield from self._get_vlan_all_ports(interface_dict, output)
+
+    def _get_vlan_all_ports(self, interface_dict, output):
+        find_regexp = r"^(\d+)\s+(\S+)\s+\S+\s+([A-Z][a-z].*)$"
+        find = re.findall(find_regexp, output, re.MULTILINE)
+        for v in find:
+            yield v[0], {
+                "name": v[1],
+                "interfaces": [
+                    interface_dict[x.strip()] for x in v[2].split(",")
+                ],
+            }
+        find_regexp = r"^(\d+)\s+(\S+)\s+\S+$"
+        find = re.findall(find_regexp, output, re.MULTILINE)
+        for v in find:
+            yield v[0], {"name": v[1], "interfaces": []}
+
+    def _get_vlan_from_id(self, interface_dict):
+        command = "show vlan brief"
+        output = self.device.cli([command])[command]
+        vlan_regexp = r"^(\d+)\s+(\S+)\s+\S+.*$"
+        find_vlan = re.findall(vlan_regexp, output, re.MULTILINE)
+        for vlan in find_vlan:
+            command = "show vlan id {}".format(vlan[0])
+            output = self.device.cli([command])[command]
+            interface_regex = r"{}\s+{}\s+\S+\s+([A-Z][a-z].*)$".format(
+                vlan[0], vlan[1]
+            )
+            interfaces = re.findall(interface_regex, output, re.MULTILINE)
+            if len(interfaces) == 1:
+                yield vlan[0], {
+                    "name": vlan[1],
+                    "interfaces": [
+                        interface_dict[x.strip()] for x in interfaces[0].split(",")
+                    ],
+                }
+            elif len(interfaces) == 0:
+                yield vlan[0], {"name": vlan[1], "interfaces": []}
+            else:
+                logger.error(
+                    "Switch %s Error parsing for vlan id %s, "
+                    "found more values than can be.",
+                    self.device.hostname, vlan[0]
+                )
+                yield None, None
```

Comment:
Same thing as before about the exception/return None in case of error.

Reply:
In theory, this is not when should not happen. I could not leave if without an else, or at least a return value.